main.js

```diff
@@ -620,6 +662,256 @@ app.post('/delete-admin', async function(req, res) {
 	}
 });
 
+app.get('/kits', async (req, res) => {
+	if (redirectToLogin(req, res)) {
+		return;
+	}
+	const filterQuery = {'_id': new mdb.ObjectId(req.session._id)};
+	const user = await db.collection('BBY-6_users').findOne(filterQuery);
+	if(!user) {
+		// Could not find user
+		res.status(500).send("userNotFound");
+		return;
+	}
+	res.status(200).send(user.kits);
+})
+
+app.post('/kits', async (req, res) => {
+	if (redirectToLogin(req, res)) {
+		return;
+	}
+	try {
+		const filterQuery = {'name': req.body.name};
+		const kitTemplate = await db.collection('BBY-6_kit-templates').findOne(filterQuery);
+		if(!kitTemplate) {
+			res.status(500).send("kitTemplateNotFound");
+			return;
+		}
+		try {
+			const userFilterQuery = {'_id': new mdb.ObjectId(req.session._id)};
+			kitTemplate.kit = kitTemplate.kit.map(element => {
+				element.required = true;
+				element.completed = false;	
+				return element;
+			})
+			const user = await db.collection('BBY-6_users').updateOne(userFilterQuery, {$push: {kits: kitTemplate}});
```

Comment:
This pattern of searching the DB for the user using the existing session ID with a `userFilterQuery`-type thing is so common that we should probably factor it out into a `getUserFromSession(req: Request<…>)` function, or maybe a `makeFilterQuery(req: Request<…>)` function.